src/MPDevice.cpp

```diff
@@ -6766,6 +6779,119 @@ bool MPDevice::checkImportedDataNodes(const MessageHandlerCb &cb, Common::DataAd
     return true;
 }
 
+bool MPDevice::checkImportedLoginLastChildNodeUsedAddr()
+{
+    if (!isBLE())
+    {
+        qCritical() << "Last child node used address is only available for BLE";
+        return false;
+    }
+
+    for (auto* loginNode : qAsConst(loginNodes))
+    {
+        // If service is not tagged for merge it will be removed, so do not need to check
+        if (loginNode->getMergeTagged())
+        {
+            auto* bleLoginNode = static_cast<MPNodeBLE*>(loginNode);
+            const auto serviceName = bleLoginNode->getService();
+            qDebug() << "Service name: " << serviceName;
+            for (int i = 0; i < importedLoginNodes.size(); ++i)
+            {
+                // Credential parent node is found in the imported list
+                if (serviceName == importedLoginNodes[i]->getService())
+                {
+                    auto* importedLoginNode = static_cast<MPNodeBLE*>(importedLoginNodes[i]);
+                    // If the imported last child node used address is set, find the same login in local list
+                    if (Common::NOT_SET_ADDR != importedLoginNode->getLastChildNodeUsedAddr())
+                    {
+                        // First find the login name of the last child node used from imported list
+                        auto* importedLastUsedChildNode = findNodeWithAddressWithGivenParentInList(importedLoginChildNodes, importedLoginNode, importedLoginNode->getLastChildNodeUsedAddr(), 0);
+                        if (importedLastUsedChildNode)
+                        {
+                            auto lastUsedChildName = importedLastUsedChildNode->getLogin();
+                            qDebug() << "Last used child name: " << lastUsedChildName;
+                            // Find the node with the same login name for the given service
+                            auto* lastUsedChild = findNodeWithLoginWithGivenParentInList(loginChildNodes, loginNode, lastUsedChildName);
+                            if (lastUsedChild && lastUsedChild->getAddress() != bleLoginNode->getLastChildNodeUsedAddr())
```

Comment:
nice! but maybe an else for the lastUsedChild condition, as it'd mean we messed up?